simulation/simple_sensor_simulator/src/simple_sensor_simulator.cpp

```diff
@@ -213,47 +214,49 @@ void ScenarioSimulator::despawnEntity(
   simulation_api_schema::DespawnEntityResponse & res)
 {
   bool found = false;
-  res = simulation_api_schema::DespawnEntityResponse();
-  std::vector<traffic_simulator_msgs::VehicleParameters> vehicles;
-  for (const auto & vehicle : vehicles_) {
-    if (vehicle.name() != req.name()) {
-      vehicles.emplace_back(vehicle);
+  auto emplaceIfRequestedEntity = [&](const auto & entity, auto & container) {
```

Comment:
I have no problem with your changes, but this member function itself was so inefficient that I rewrote it. Please replace the implementation of this member function.


```
void ScenarioSimulator::despawnEntity(
  const simulation_api_schema::DespawnEntityRequest & req,
  simulation_api_schema::DespawnEntityResponse & res)
{
  auto remove_despawn_requested_entity_from = [&](auto & v) {
    const auto size = std::size(v);
    v.erase(
      std::remove_if(
        std::begin(v), std::end(v),
        [&](const auto & entity) { return entity.name() == req.name(); }),
      std::end(v));
    return size != std::size(v);  // true if something removed.
  };

  const auto ego_entity_was_removed = remove_despawn_requested_entity_from(ego_vehicles_);

  if (ego_entity_was_removed) {
    ego_entity_simulation_.reset();
  }

  const auto any_entity_was_removed = ego_entity_was_removed or
                                      remove_despawn_requested_entity_from(vehicles_) or
                                      remove_despawn_requested_entity_from(pedestrians_) or
                                      remove_despawn_requested_entity_from(misc_objects_);

  if (any_entity_was_removed) {
    entity_status_.erase(req.name());
  }

  res.mutable_result()->set_success(any_entity_was_removed);
}
```

Reply:
Indeed, creating additional vectors did not make sense, I myself didn't want to change the previous implementation so much. I will try to pay more attention to code optimization. Thank you! ([commit](https://github.com/tier4/scenario_simulator_v2/pull/1046/commits/3e26c62cb63ace0794f4794fceb8c9911934ef39))